### ISSUE TITLE
Unable to open '/frameworks/locale/.../MyApplication.mxml'

### ISSUE THREAD

**Issue by velara3** [BowlerHatLLC/vscode-as3mxml#356]
I'm farther along but I've now encountered an error after import of a previous project.

The problems view has this error:

>/SDKs/Flex_SDK/Flex_4.16.1_AIR_32.0/frameworks/locale/Users/user/Documents/projects/MyProject/src/MyApplication.mxml'. /Users/user/Downloads/SDKs/Flex_SDK/Flex_4.16.1_AIR_32.0/frameworks/air-config.xml (line: 85)

air-config.xml line 85 is the last entry here: 

      
      <library-path>
         <path-element>libs</path-element>
         <path-element>libs/mx</path-element>
         <path-element>libs/air</path-element>
         <path-element>libs/air</path-element>
         <path-element>locale/{locale}</path-element>
      </library-path>

And my asconfig.json is: 

    {
	"compilerOptions": {
		"output": "bin-debug/MyApplication.swf",
		"accessible": false,
		"source-path": [
			"src"
		],
		"library-path": [
			"libs"
		]
	},
	"files": [
		"src/MyApplication.mxml"
	],
	"application": "src/MyApplication-app.xml",
	"copySourcePathAssets": true,
	"additionalOptions": "-locale en_US",
	"airOptions": {
		"output": "MyApplication.air",
		"signingOptions": {
			"keystore": "...",
			"storetype": "..."
		}
	},
	"config": "air"
    }

Have you seen this error before?

**Comment by velara3** [comment]
If I remove the additionalOptions line the error goes away. 

	"additionalOptions": "-locale en_US",

**Comment by devlfm** [comment]
Hi, I have the same error, but I can't remove locale.
`"additionalOptions": "-locale=pt_PT -allow-source-path-overlap=true -source-path=locale/{locale}"`

> Elapsed time: 0.22642595 seconds
	at flex2.compiler.common.CompilerConfiguration.addPathElementToListAsVirtualFile(CompilerConfiguration.java:231)
	at flex2.compiler.common.CompilerConfiguration.expandLocaleToken(CompilerConfiguration.java:197)
	at flex2.compiler.common.CompilerConfiguration.expandTokens(CompilerConfiguration.java:125)
	at flex2.compiler.common.CompilerConfiguration.cfgSourcePath(CompilerConfiguration.java:1904)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at flex2.compiler.config.ConfigurationBuffer.commitVariable(ConfigurationBuffer.java:695)
	at flex2.compiler.config.ConfigurationBuffer.commit(ConfigurationBuffer.java:644)
	at flex2.tools.Mxmlc.processConfiguration(Mxmlc.java:706)
	at flex2.tools.Mxmlc.processConfiguration(Mxmlc.java:559)
	at flex2.tools.Fcsh.mxmlc(Fcsh.java:835)
	at flex2.tools.Fcsh.process(Fcsh.java:141)
	at flex2.tools.Fcsh.main(Fcsh.java:78)
command line: Error: unable to open 'locale/pt_PT'

**Comment by devlfm** [comment]
@joshtynjala can you see closed issues?

**Comment by joshtynjala** [comment]
> at flex2.tools.Fcsh.main(Fcsh.java:78)
> command line: Error: unable to open 'locale/pt_PT'

This error seems to be when you are compiling.

It may be related to `-source-path=locale/{locale}`. You should be using += there because = will remove all other source-path entries.

**Comment by devlfm** [comment]
`"additionalOptions"` were defined by the auto-import feature from FlashBuilder project.

I changed to `"-locale=pt_PT -allow-source-path-overlap=true -source-path+=locale/{locale}"` and the result was the same.

I noticed that the output from compiler had 2 dashes `--` before each parameter:
> (fcsh) mxmlc --debug=false --output=bin-debug/main.swf --accessible=true --source-path+=src --library-path+=libs -locale=pt_PT -allow-source-path-overlap=true -source-path+=locale/{locale} -- src/main.mxml

So I changed asconfig.json also to match:
 `"--locale=pt_PT --allow-source-path-overlap=true --source-path+=locale/{locale}"`

But keeps throwing same error.
Then I had set full path to the locale folder and compiled!
I also tried to remove the '+=' and worked too, I guess is because of the -allow-source-path-overlap=true`

In conclusion, this works for me: `-locale=pt_PT -allow-source-path-overlap=true -source-path=C:/app/src/locale/{locale}` 

I hope it helps you to fix

**Comment by joshtynjala** [comment]
@devlfm I'm guessing that this might also work, if you prefer a relative path:

```
-source-path=src/locale/{locale}
```

**Comment by devlfm** [comment]
Yes it works! thank you josh.. maybe this should be added to documentation to help others that use the auto-import feature.
